prov/sockets/src/sock_progress.c

```diff
@@ -2520,10 +2517,11 @@ static void sock_pe_wait(struct sock_pe *pe)
 {
 	char tmp;
 	int ret;
+	void *ep_contexts[SOCK_EPOLL_WAIT_EVENTS];
 
-	ret = sock_epoll_wait(&pe->epoll_set, -1);
-        if (ret < 0)
-                SOCK_LOG_ERROR("poll failed : %s\n", strerror(errno));
+	ret = fi_epoll_wait(pe->epoll_set, ep_contexts, SOCK_EPOLL_WAIT_EVENTS, -1);
```

Comment:
We only need to block until at least one underlying socket is ready.  We can drop the array and request only a single event.